pkg/translator/annotations.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright © 2022 Kubernetes Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package translator
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	networkingv1 "k8s.io/api/networking/v1"
+)
+
+type IngressProvider string
+
+const (
+	IngressNginxIngressProvider IngressProvider = "ingress-nginx"
+)
+
+type ProviderPrefix string
+
+const (
+	IngressNginxProviderPrefix ProviderPrefix = "nginx.ingress.kubernetes.io"
```

Comment:
Unfortunately not all implementations are going to have a consistent annotation prefix, for example https://cloud.google.com/kubernetes-engine/docs/how-to/load-balance-ingress#summary_of_external_ingress_annotations.